### ISSUE TITLE
Set up to work in pure browser-based environment

### ISSUE THREAD

**Issue by misaugstad** [rogierschouten/async-lock#66]
If you just try to load the script from a CDN and you're not using Node, you'll get an error message:
>Uncaught ReferenceError: module is not defined

I think that all we'd need to do is wrap the `module.exports = AsyncLock;` in something that checks if module is defined. I think it's standard to do something like
```
if ("object"==typeof exports && "undefined" != typeof module) {
    module.exports = functions;
}
```

**Comment by rogierschouten** [comment]
As stated in the readme I will only accept PRs and not do any further development on this package. 
I took it over from the author but it's a mess and I don't use it myself. 
Feel free to send a PR provided it has tests for your change.

**Comment by misaugstad** [comment]
Yep! Sorry, didn't mean to imply that you would do the work, just logging it as an issue for now. I don't feel confident that I have the knowledge to make this change while ensuring that I don't break something for others. So I'll either come back to this later, or hopefully someone else sees this and is willing to take it on, since it should be really simple if you know what you're doing!
